From da1f09f2a226dc5afc9bb4d1fbdbfe459d5f9970 Mon Sep 17 00:00:00 2001
From: William Wu <william.wu@rock-chips.com>
Date: Mon, 8 Apr 2019 15:58:26 +0800
Subject: [PATCH] dt-bindings: usb-xhci: add xhci-trb-ent-quirk property

This patch adds a new "xhci-trb-ent-quirk" property for some
Rockchip platforms which need to enable the ENT flag in the
TRB to force the xHC to pre-fetch the next TRB of a TD.

Change-Id: I670cfc759433b858feb1d5bb2805c793b050328a
Signed-off-by: William Wu <william.wu@rock-chips.com>
---
 Documentation/devicetree/bindings/usb/usb-xhci.txt | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/Documentation/devicetree/bindings/usb/usb-xhci.txt b/Documentation/devicetree/bindings/usb/usb-xhci.txt
index dab51612eb8f..d520675ac44a 100644
--- a/Documentation/devicetree/bindings/usb/usb-xhci.txt
+++ b/Documentation/devicetree/bindings/usb/usb-xhci.txt
@@ -38,6 +38,9 @@ Optional properties:
   - quirk-broken-port-ped: set if the controller has broken port disable mechanism
   - xhci-slow-suspend: set if it needs an extraordinary delay to wait
     for xHC enter the Halted state after the Run/Stop (R/S) bit is cleared to '0'
+  - xhci-trb-ent-quirk: set if this xhci platform need to enable the Evaluate
+    Next TRB(ENT) flag in the TRB data structure to force xHC to pre-fetch
+    the next TRB of a TD.
   - imod-interval-ns: default interrupt moderation interval is 5000ns
   - phys : see usb-hcd.txt in the current directory
 
-- 
2.35.3

